test/integration/helpers_test.go

```diff
@@ -0,0 +1,330 @@
+
+	tc := &TestContext{
+		Ctx:     ctx,
+		T:       t,
+		TestID:  testID,
+		Service: svc,
+		Deps:    deps,
+	}
+
+	t.Cleanup(func() {
+		tc.Cleanup()
+	})
+
+	return tc
+}
+
+// Cleanup removes all test data tagged with this test ID
+func (tc *TestContext) Cleanup() {
+	_, _ = tc.Service.ExecuteWriteQuery(
+		context.Background(),
+		"MATCH (n) WHERE n.test_id = $testID DETACH DELETE n",
+		map[string]any{"testID": tc.TestID},
+		sharedCfg.Database,
+	)
+}
+
+// SeedNode creates a test node and returns it
+func (tc *TestContext) SeedNode(label string, props map[string]any) error {
+	tc.T.Helper()
+
+	// Always add test_id to props
+	props["test_id"] = tc.TestID
+
+	session := (sharedDrv).NewSession(tc.Ctx, neo4j.SessionConfig{})
+	defer session.Close(tc.Ctx)
+
+	query := fmt.Sprintf("CREATE (n:%s $props) RETURN n", label)
+	_, err := session.ExecuteWrite(tc.Ctx, func(tx neo4j.ManagedTransaction) (any, error) {
+		result, err := tx.Run(tc.Ctx, query, map[string]any{"props": props})
+		if err != nil {
+			return nil, err
+		}
+		_, err = result.Collect(tc.Ctx)
+		return nil, err
+	})
+
+	return err
+}
+
+// CallTool invokes an MCP tool and returns the response
+func (tc *TestContext) CallTool(handler func(context.Context, mcp.CallToolRequest) (*mcp.CallToolResult, error), args map[string]any) *mcp.CallToolResult {
+	tc.T.Helper()
+
+	req := mcp.CallToolRequest{
+		Params: mcp.CallToolParams{
+			Arguments: args,
+		},
+	}
+
+	res, err := handler(tc.Ctx, req)
+	if err != nil {
+		tc.T.Fatalf("tool call failed: %v", err)
+	}
+	if res == nil {
+		tc.T.Fatal("tool returned nil response")
+	}
+	if res.IsError {
+		tc.T.Fatalf("tool returned error: %+v", res)
+	}
+
+	return res
+}
+
+// ParseJSONResponse parses JSON response into the provided interface
+func (tc *TestContext) ParseJSONResponse(res *mcp.CallToolResult, v any) {
+	tc.T.Helper()
+
+	if len(res.Content) == 0 {
+		tc.T.Fatal("response has no content")
+	}
+
+	textContent, ok := mcp.AsTextContent(res.Content[0])
+	if !ok {
+		tc.T.Fatalf("expected TextContent, got %T", res.Content[0])
+	}
+
+	if err := json.Unmarshal([]byte(textContent.Text), v); err != nil {
+		tc.T.Fatalf("failed to parse JSON response: %v\nraw: %s", err, textContent.Text)
+	}
+}
+
+// VerifyNodeInDB verifies that a node exists in the database with the given properties
+func (tc *TestContext) VerifyNodeInDB(label string, props map[string]any) *neo4j.Record {
+	tc.T.Helper()
+
+	// Build WHERE clause for each property
+	props["test_id"] = tc.TestID
+
+	session := (sharedDrv).NewSession(tc.Ctx, neo4j.SessionConfig{})
+	defer session.Close(tc.Ctx)
+
+	// Build WHERE clause dynamically
+	whereClauses := []string{}
+	for key := range props {
+		whereClauses = append(whereClauses, fmt.Sprintf("n.%s = $%s", key, key))
+	}
+	whereClause := ""
+	if len(whereClauses) > 0 {
+		whereClause = " WHERE " + whereClauses[0]
+		for i := 1; i < len(whereClauses); i++ {
+			whereClause += " AND " + whereClauses[i]
+		}
```

Comment:
we can simplify it via something like:
```suggestion
		whereClause = " WHERE " + strings.Join(whereClauses, " AND ")
```